cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/terminology/R4FileSystemFhirTerminologyProvider.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.cohort.cql.terminology;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.cqframework.cql.elm.execution.VersionedIdentifier;
+import org.hl7.fhir.r4.model.ValueSet;
+import org.hl7.fhir.r4.model.ValueSet.ConceptReferenceComponent;
+import org.hl7.fhir.r4.model.ValueSet.ConceptSetComponent;
+import org.opencds.cqf.cql.engine.runtime.Code;
+import org.opencds.cqf.cql.engine.terminology.CodeSystemInfo;
+import org.opencds.cqf.cql.engine.terminology.ValueSetInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ca.uhn.fhir.context.ConfigurationException;
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.parser.DataFormatException;
+
+/**
+ * This class is used to read ValueSet definitions from the local filesystem or S3 compatible source.
+ * ValueSet definitions are expected to be stored in FHIR xml or JSON format and can be generated from
+ * spreadsheets using the com.ibm.cohort.tooling.fhir.ValueSetImporter tool.
+ * 
+ * In order to use ValueSets stored on the filesystem, the following spark config parameters must be provided:
```

Comment:
I don't think these docs are correct. It should say "in order to use ValueSets stored in an S3 compatible endpoint" and then the properties section needs to be updated. Hadoop uses the `spark.hadoop.fs.s3a` configuration properties first. If they aren't provided, it will fall back to using the common AWS_XXX environment variables. The two settings that can be provided as environment variables are `AWS_ACCESS_KEY_ID` (vs. `-Dspark.hadoop.fs.s3a.access.key`) and `AWS_SECRET_ACCESS_KEY` (vs. `-Dspark.hadoop.fs.s3a.secret.key`). AWS_ACCESS_KEY should be updated in the docs as shown above and AWS_ENDPOINT and AWS_LOCATION should be removed as those were specific to the Spike work and they are no longer used anywhere in our code.

Reply:
Good suggestion will change